engine/gamesys/proto/gamesys/model_ddf.proto

```diff
@@ -9,24 +9,39 @@ option java_outer_classname = "ModelProto";
 
 // TODO: Add documentation for messages, see https://github.com/defold/extension-spine/blob/main/defold-spine/commonsrc/spine_ddf.proto
 
+message Texture
+{
+    required string sampler     = 1; // the name of the sampler
+    required string texture     = 2 [(resource)=true];
+}
+
+message Material
+{
+    required string name        = 1; // the name of the material in the model file
+    required string material    = 2 [(resource)=true];
+    repeated Texture textures   = 3;
+}
+
+// The source format (.model)
 message ModelDesc
 {
     required string mesh        = 2 [(resource)=true];
-    required string material    = 3 [(resource)=true];
-    repeated string textures    = 4 [(resource)=true];
+    optional string material    = 3 [(resource)=true]; // Deprecated
+    repeated string textures    = 4 [(resource)=true]; // Deprecated
     optional string skeleton    = 5 [(resource)=true];
     optional string animations  = 6 [(resource)=true];
     optional string default_animation  = 7;
 
     optional string name        = 10; // Deprecated
+    repeated Material materials = 11;
 }
 
+// The engine format (.modelc)
 message Model
 {
-    required string rig_scene   = 1 [(resource)=true];
-    required string material    = 2 [(resource)=true];
-    repeated string textures    = 3 [(resource)=true];
-    optional string default_animation  = 4;
+    required string rig_scene   = 1;
```

Comment:
Was it intentional to remove the `[(resource)=true]` tag from the `rig_scene` field here?

Reply:
I think so, yes.
Not that it's a huge deal, but we never look at those properties at runtime. It is a string, and should already be translated at that point.

Reply:
You mean that the editor parses metadata from the engine formats?
That would be unfortunate ofc (no unit tests has complained) but I can put it back.